### ISSUE TITLE
GPU filtered back projection FBP_CUDA and CPU FBP results in different values

### ISSUE THREAD

**Issue by teracamo** [astra-toolbox/astra-toolbox#144]
Seems like the values result from FBP_CUDA and FBP algorithms are different. The CPU algorithm should be the one returning more reasonable values

**Comment by dmpelt** [comment]
I think in principle it is expected that they give slightly different results, since the projection operation is defined slightly different in both (and in the CPU case, this definition can be changed with `line`, `linear` etc). However, in practice the difference shouldn't be large -- do you have an example of this? If so, there might be a bug that we have to fix...

**Comment by teracamo** [comment]
Well the difference were quite significant, I suspect some normalization were missed for the CUDA FBP, plus the image coordinate were flipped. I also forgot to mention this happens for line_fanflat projector geometry and fanflat projection geometry.

Here is the code you can use to test, I used version 1.8
```python
import astra
import numpy as np
import matplotlib as mpl
mpl.use('Qt5Agg')
import matplotlib.pyplot as plt

def main():
    phantom = np.zeros((128, 128))
    phantom[32 : 64, 32 : 64] = np.ones((32, 32)) * 1000
    phantom[64 : 96, 64 : 96] = np.ones((32, 32)) * -400

    vol_geom = astra.create_vol_geom(
        128, 128
    )

    fanp_geom = astra.create_proj_geom('fanflat',
                                       1,
                                       256,
                                       np.linspace(-np.pi, np.pi, 512),
                                       512,
                                       256
                                       )
    vol_id = astra.data2d.create('-vol', vol_geom)
    rec_id = astra.data2d.create('-vol', vol_geom)
    cuda_rec_id = astra.data2d.create('-vol', vol_geom)
    proj_id = astra.create_projector('line_fanflat', fanp_geom, vol_geom)
    cuda_proj_id = astra.create_projector('cuda', fanp_geom, vol_geom)
    astra.data2d.store(vol_id, phantom)

    ## Projection ##
    [sino_id, sino] = astra.create_sino(vol_id, proj_id)
    [cuda_sino_id, sino] = astra.create_sino(vol_id, cuda_proj_id)

    ## FBP ##
    cfg = astra.astra_dict('FBP')
    cfg['ProjectionDataId'] = sino_id
    cfg['ReconstructionDataId'] = rec_id
    cfg['ProjectorId'] = proj_id

    algo_id = astra.algorithm.create(cfg)
    astra.algorithm.run(algo_id)

    ## CUDA FBP ##
    cfg = astra.astra_dict('FBP_CUDA')
    cfg['ProjectionDataId'] = sino_id
    cfg['ReconstructionDataId'] = cuda_rec_id
    cfg['ProjectorId'] = cuda_proj_id # I have also tried using non-cuda proj_id too, same results.

    algo_id = astra.algorithm.create(cfg)
    astra.algorithm.run(algo_id)

    rec = astra.data2d.get(rec_id)
    cuda_rec = astra.data2d.get(cuda_rec_id)


    fig = plt.figure()
    ax1 = fig.add_subplot(211)
    ax2 = fig.add_subplot(212)
    ax1.imshow(rec)
    ax2.imshow(cuda_rec)
    plt.show()

    pass


if __name__ == '__main__':
    main()
```

**Comment by dmpelt** [comment]
Ah, I'm personally not very familiar with the fan-beam code, since I mostly use parallel-beam myself. But I'm able to reproduce the problem on my machine (thanks for the example code -- extremely useful!).

@wjp do you know why this is happening?

**Comment by teracamo** [comment]
You are very welcomed, this actually haunts me for a long time. I verified the contrast can be corrected with a constant scaling ~500, but I am not sure whether this scaling is constant for all images, its hard to map the pixels for real medical images. I also don't understands why it is flipped.

**Comment by wjp** [comment]
Thanks for the report. The flip has been fixed already recently, and this is now ok in git master[*]. I have a fix for the value scaling as part of a larger set of updates to fan beam FBP and cone beam FDK, which will land in master soon if I don't run into any unexpected regressions. With this fix your sample code produces the right values and orientation.

[*] If you are using conda, you can get experimental packages for git master using 
```conda install -c astra-toolbox/label/dev astra-toolbox```

**Comment by teracamo** [comment]
Thanks for the updates, would you mind tagging the commit that fix the scaling when its ready?

**Comment by wjp** [comment]
This branch is now merged into master by 87304c5e3da38d51a3f7754d1c0cbca066e9183f .
